### DISCUSSION TITLE
How to interact with the CPS fund

### DISCUSSION THREAD

**Discussion by flowersinthesand** [icon-project/btp#23] [Q&A] [answered]
Hi @BennyOptions

From https://docs.google.com/document/d/1vDQvi_sIo4LFNN0n84rPGDjxEhaNjOfV/edit

> you should call the proper method to increase the size of the CPS fund you should call the proper method to increase the size of the CPS fund

What's that proper method?

Also is there anything that we should be aware when interacting with the CPS fund?

**Comment by BennyOptions** [comment, marked as answer]
I am asking the CPS developers and will get back to you within 1-2 days

**Reply by BennyOptions** [reply]
The method is add_fund() @flowersinthesand and it's the CPF_Treasury SCORE (cxdca1178010b5368aea929ad5c06abee64b91acc2) on main net

https://tracker.icon.foundation/contract/cxdca1178010b5368aea929ad5c06abee64b91acc2#readcontract

**Reply by flowersinthesand** [reply]
Thanks for your response @BennyOptions

As we are not familiar to how CPS works at the moment I'd like to make sure our thought is correct before implementing it. Is it right to call add_fund with the amount of ICX to be sent to the CPS and then send it to the CPS every time the Fee Aggregation contract tries to send ICX to the CPS? Or is calling add_fund conditional by some condition?

**Reply by aceraizel** [reply]
Hi @BennyOptions, Will I be able to generate this CPS address for the local node for test?

**Reply by BennyOptions** [reply]
I am talking with the team to see if you can get access to their repo. However, I provided the SCORE address and all the information you ask here should be in the codes right? 

For add_fund(), you should be able to call this everytime you send ICX to the fund. 

For deploying to local node, is it not possible to copy the codes from the contract above and deploy locally?

I have also asked for access to their repo so you can get more info, but since it is a smart contract I believe the data is available via checking the SCORE codes

**Reply by BennyOptions** [reply]
Here is the repo

https://github.com/ibriz/CPS/tree/master/

**Reply by aceraizel** [reply]
Thank you @BennyOptions 
This repo helps me so much. I will try to deploy this smart contract to the local node to test.

**Reply by aceraizel** [reply]
Hi @BennyOptions , I have tried to deploy this CPF to the local node (build from [here](https://github.com/icon-project/goloop/blob/master/doc/gochain_icon_local_node_guide.md))

```
git clone https://github.com/ibriz/CPS/tree/master
cd CPS/CPFTreasury
zip -r -v ./cpf_treasury.zip cpf_treasury lib package.json
cp ./cpf_treasury.zip ${GOCHAIN_LOCAL_ROOT}/

cd ${GOCHAIN_LOCAL_ROOT}
./goloop rpc sendtx deploy ./cpf_treasury.zip --uri http://localhost:9082/api/v3 --key_store ./data/godWallet.json --key_password gochain --nid 3 --step_limit 10000000000 --content_type application/zip
```

Then I have received this error 
![image](https://user-images.githubusercontent.com/45514397/114988275-590f0c80-9ec0-11eb-80a1-93a4cf2c0df9.png)

Am I missing something? Could you please give me the guide to deploying this contract?

**Reply by flowersinthesand** [reply]
@aceraizel Please open another discussion for that question.

**Reply by flowersinthesand** [reply]
It looks like a CPS issue. I think it's better to raise an issue in their repository and ask Scott to push it.

**Reply by aceraizel** [reply]
The error screen has been gotten when deploying to the local node. So, I think maybe not only CPS issue, but also ICON SCORE Service
